yaki_mobile/lib/presentation/state/notifiers/declaration_notifier.dart

```diff
@@ -31,14 +32,81 @@ class DeclarationNotifier extends StateNotifier<String> {
   ///
   /// create a DeclarationModel model instance,
   ///
-  /// then invoke the declarationRepository.create(), that will send the newly declaration to the API (via _api.dart)
-  Future<void> create(String status) async {
+  /// then invoke the declarationRepository.createAllDay(), that will send the newly declaration to the API (via _api.dart)
+  Future<void> createAllDay(String status) async {
+    final todayDate = DateTime.now();
+
     DeclarationModel newDeclaration = DeclarationModel(
-      declarationDate: DateTime.now(),
+      declarationDate: todayDate,
+      declarationDateStart: DateTime.parse(
+        '${parseDate(todayDate)} 00:00:00Z',
+      ),
+      declarationDateEnd: DateTime.parse(
+        '${parseDate(todayDate)} 23:59:59Z',
+      ),
```

Comment:
be careful with timezones. why do you hard code zulu timezone ?

should be better to use the system timezone instead